src/App/Fossa/Ficus/Analyze.hs

```diff
@@ -187,45 +179,45 @@ runFicus ficusConfig = do
+    --     <> pretty (length $ ficusMessageErrors messages)
+    --     <> " errors, "
+    --     <> pretty (length $ ficusMessageDebugs messages)
+    --     <> " debug messages, "
+    --     <> pretty (length $ ficusMessageFindings messages)
+    --     <> " findings"
 
-    pure messages
+    pure result
   where
-    streamFicusOutput :: Handle -> IO FicusMessages
-    streamFicusOutput handle = do
-      let loop acc = do
-            eof <- hIsEOF handle
-            if eof
-              then do
-                putStrLn "[DEBUG] Reached end of Ficus output stream"
-                pure acc
-              else do
-                line <- hGetLine handle
-                let lineBS = BL.fromStrict $ Text.Encoding.encodeUtf8 $ toText line
-                case decode lineBS of
-                  Just message -> do
-                    -- Log messages as they come, with timestamps
-                    now <- getCurrentTime
-                    let timestamp = formatTime defaultTimeLocale "%H:%M:%S.%3q" now
-                    case message of
-                      FicusMessageError err ->
-                        putStrLn $ "[" ++ timestamp ++ "] ERROR " <> toString (displayFicusError err)
-                      FicusMessageDebug dbg ->
-                        putStrLn $ "[" ++ timestamp ++ "] DEBUG " <> toString (displayFicusDebug dbg)
-                      FicusMessageFinding finding ->
-                        putStrLn $ "[" ++ timestamp ++ "] FINDING " <> toString (displayFicusFinding finding)
-                    let newMessage = singletonFicusMessage message
-                    loop (acc <> newMessage)
-                  Nothing -> do
-                    loop acc
-      loop mempty
+    currentTimeStamp :: IO String
+    currentTimeStamp = do
+      now <- getCurrentTime
+      pure . formatTime defaultTimeLocale "%H:%M:%S.%3q" $ now
+
+    streamFicusOutput :: Handle -> IO (Maybe FicusSnippetScanResults)
+    streamFicusOutput handle =
+      Conduit.runConduit $
+        CC.sourceHandle handle
+          .| CC.decodeUtf8Lenient
+          .| CC.linesUnbounded
+          .| CCL.mapMaybe decodeStrictText
+          .| CC.foldM
+            ( \acc message -> do
+                -- Log messages as they come, with timestamps
+                timestamp <- currentTimeStamp
+                case message of
+                  FicusMessageError err -> do
+                    putStrLn $ "[" ++ timestamp ++ "] ERROR " <> toString (displayFicusError err)
+                    pure acc
+                  FicusMessageDebug dbg -> do
+                    putStrLn $ "[" ++ timestamp ++ "] DEBUG " <> toString (displayFicusDebug dbg)
+                    pure acc
+                  FicusMessageFinding finding -> do
+                    putStrLn $ "[" ++ timestamp ++ "] FINDING " <> toString (displayFicusFinding finding)
+                    pure $ acc <|> (FicusSnippetScanResults <$> findingToAnalysisId finding)
```

Comment:
I think it's an error if we find more than one log line that returns non-Nothing from `findingToAnalysisId`. Or at least something we want to log as suspicious.

Can we add something here that logs that?

Reply:
Yep, I made a `when` block that will do that.